internalsdk/pro/pro.go

```diff
@@ -53,29 +53,28 @@ type ProClient interface {
 
 // NewClient creates a new instance of ProClient
 func NewClient(baseURL string, opts *webclient.Opts) ProClient {
-	httpClient := opts.HttpClient
-	if httpClient == nil {
-		httpClient = &http.Client{}
-	}
-	client := &proClient{
+	httpClient := GetHTTPClient(opts)
+	wc := webclient.NewRESTClient(defaultwebclient.SendToURL(httpClient, baseURL,
+		func(client *resty.Client, req *http.Request) error {
+			prepareProRequest(req, common.ProAPIHost, opts.UserConfig())
+			return nil
+		}, nil))
+	return &proClient{
 		userConfig: opts.UserConfig,
+		webclient:  wc,
```

Comment:
I like the option where we can pass transport rather than it uses the same for all, I think what Harry meant is for user API we can use  `ChainedNonPersistent` but for other, we can still use `ParallelForIdempotent`

Reply:
I wasn't clear on that, but I just pushed a change to continue using ParallelForIdempotent in the ProClient